mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,462 @@
+        logged is given by `num_eval_images`. Please refer to
+        https://docs.wandb.ai/guides/data-vis to learn more about W&B Tables.
+        Currently, the `WandbLogger` logs the predicted bounding boxes along
+        with the ground truth at every evaluation interval.
+        Note: This depends on the `EvalHook` whose priority is more than
+        `WandbLogger`. Also note that the data is just logged once and
+        subsequent evaluation tables uses reference to the logged data to save
+        memory usage.
+
+    ```
+    Example:
+        log_config = dict(
+            interval=10,
+            hooks=[
+                dict(type='WandbLogger',
+                     wandb_init_kwargs={
+                         'entity': WANDB_ENTITY,
+                         'project': WANDB_PROJECT_NAME
+                     },
+                     logging_interval=10,
+                     log_checkpoint=True,
+                     log_checkpoint_metadata=True,
+                     num_eval_images=100)
+            ])
+    ```
+
+    Args:
+        wandb_init_kwargs (dict): A dict passed to wandb.init to initialize
+            a W&B run. Please refer to https://docs.wandb.ai/ref/python/init
+            for possible key-value pairs.
+        logging_interval (int): Logging interval (every k iterations).
+            Default 10.
+        log_checkpoint (bool): Save the checkpoint at every checkpoint interval
+            as W&B Artifacts. Use this for model versioning where each version
+            is a checkpoint.
+            Default: False
+        log_checkpoint_metadata (bool): Log the evaluation metrics computed
+            on the validation data with the checkpoint, along with current
+            epoch as a metadata to that checkpoint.
+            Default: True
+        num_eval_images (int): Number of validation images to be logged.
+            Default: 100
+    """
+
+    def __init__(self,
+                 wandb_init_kwargs=None,
+                 interval=10,
+                 log_checkpoint=False,
+                 log_checkpoint_metadata=False,
+                 num_eval_images=100,
+                 **kwargs):
+        super(WandbLogger, self).__init__(wandb_init_kwargs, interval,
+                                          **kwargs)
+
+        self.log_checkpoint = log_checkpoint
+        self.log_checkpoint_metadata = log_checkpoint_metadata
+        self.num_eval_images = num_eval_images
+        self.log_eval_metrics = True
+        self.best_score = 0
+        self.val_step = 0
+
+    @master_only
+    def before_run(self, runner):
+        super(WandbLogger, self).before_run(runner)
+        self.cfg = self.wandb.config
+        # Check if configuration is passed to wandb.
+        if len(dict(self.cfg)) == 0:
+            warnings.warn(
+                'To log mmdetection Config, '
+                'pass it to init_kwargs of WandbLogger.', UserWarning)
```

Comment:
The config in mmdetection is too huge for users to be added by `init_kwargs`. Is it possible to add config from the file?

Reply:
By adding config from file, do you mean log something like this - here's an example command line:

`python tools/train.py configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py`

Do you want to log the config from `configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py`? It's doable in my opinion.